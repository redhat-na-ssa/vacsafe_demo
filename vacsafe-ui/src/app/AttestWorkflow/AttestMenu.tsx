import * as React from 'react';
import {
  Button,
  Card,
  CardBody,
  CardFooter,
  Divider,
  Stack,
  Text,
} from "@patternfly/react-core";
import { FileUploadIcon, CheckCircleIcon } from '@patternfly/react-icons';
import { useHistory } from 'react-router-dom';
import CardWithTitle from '@app/Components/CardWithTitle';
import { useKeycloak } from "@react-keycloak/web";
import { useState  } from 'react';

const AttestMenu: React.FunctionComponent = () => {
    const history = useHistory();
    const { keycloak } = useKeycloak();
    const [employeeRole, setEmployeeRole] = useState(""); 

    if(keycloak?.authenticated) {
      keycloak.loadUserProfile().then(() => {
        if( keycloak.hasRealmRole("AG-COVIDSafe-Approver") || keycloak.hasRealmRole("AG-COVIDSafe-Approver-Test")){
          setEmployeeRole("approver"); 
        }  
      })
    }

    return (
      <Card isRounded>
        <CardWithTitle title="Menu" info="Please select from the following:" />
        <CardBody>
          <Stack hasGutter>
            <Button
              icon={<FileUploadIcon />}
              id="submit-vax-card-menu-button"
              isSmall
              onClick={() => history.push("/attest/vax")}
              variant="tertiary"
            >
              <Text>Submit Vaccination Record</Text>
              <Text>Supporting Document Upload Required</Text>
            </Button>

            <Button
              icon={<FileUploadIcon />}
              id="submit-vax-test-menu-button"
              isSmall
              onClick={() => history.push("/attest/test")}
              variant="tertiary"
            >
              <Text>Submit COVID Test Results</Text>
              <Text>Supporting Document Upload Required</Text>
            </Button>
            {employeeRole.localeCompare("approver") == 0 && (
              <>
                <Divider />
                <Button
                  icon={<CheckCircleIcon />}
                  id="validate-submissions-button"
                  isSmall
                  onClick={() => history.push("/hrpanels")}
                  variant="tertiary"
                >
                  <Text>HR Dashboard</Text>
                  <Text>Review or Update Red Hat VCS Records</Text>
                </Button>
              </>
            )}
          </Stack>
        </CardBody>
        <CardFooter>
          <Button variant="plain" onClick={() => history.push("/profile")}>
            Back
          </Button>
        </CardFooter>
      </Card>
    );
};
export { AttestMenu }
