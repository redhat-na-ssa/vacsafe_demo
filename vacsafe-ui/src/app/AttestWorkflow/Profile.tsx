import React, { useState } from "react";
import {
  Button,
  Card,
  CardBody,
  Checkbox,
  Form,
  FormGroup,
  TextInput,
} from "@patternfly/react-core";
import { CheckCircleIcon } from "@patternfly/react-icons";
import CardWithTitle from "@app/Components/CardWithTitle";
import { useForm, Controller } from "react-hook-form";
import { useHistory } from "react-router-dom";
import { dateConvert } from "@app/utils/utils";
import { useKeycloak } from "@react-keycloak/web";
import BirthDatePicker from "@app/Components/BirthDatePicker";

const Profile: React.FunctionComponent = () => {
  const history = useHistory();
  const { control, handleSubmit } = useForm();
  const { keycloak } = useKeycloak();
  
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [employeeAgency, setEmployeeAgency] = useState("");
  const [employeeRole, setEmployeeRole] = useState(""); 

  if(keycloak?.authenticated) {
    keycloak.loadUserProfile().then((profile) => {
      setEmployeeId(profile.attributes.workforceid[0]);
      setEmployeeName(profile.firstName + " " + profile.lastName);
      setEmployeeEmail(profile.email || "");
      setEmployeeAgency(profile.attributes.agency[0]); 
      if( keycloak.hasRealmRole("AG-COVIDSafe-Approver") || keycloak.hasRealmRole("AG-COVIDSafe-Approver-Test")){
        setEmployeeRole("approver"); 
      }  
    })
  }

  const onSubmit = data => {
    sessionStorage.setItem("alternateEmail", data["alternateEmail"]);
    sessionStorage.setItem("dateOfBirth", dateConvert(data["dateOfBirth"]));
    history.push("/attestmenu");
  };

  return (
    <Card isRounded>
      <CardWithTitle
        title="Profile"
        info="The information below is retrieved from your user account."
      />
      <CardBody>
        <Form onSubmit={handleSubmit(onSubmit)}>
          <FormGroup
            id="employee-id-display"
            label="Person ID"
            fieldId="profile-form-employee-id"
          >
            <TextInput
              aria-label="employee-id-display"
              isDisabled
              type="text"
              value={employeeId}
            />
          </FormGroup>
          <FormGroup
            id="employee-full-name"
            label="Full Name"
            fieldId="profile-form-employee-full-name"
          >
            <TextInput
              aria-label="employee-full-name"
              isDisabled
              type="text"
              value={employeeName}
            />
          </FormGroup>
          <BirthDatePicker
            control={control}
            defaultBirthDate={dateConvert(sessionStorage.getItem("dateOfBirth") || "")} 
          />
          <FormGroup
            id="employee-email-display"
            label="Email"
            fieldId="profile-form-email"
          >
            <TextInput
              aria-label="employee-email-display"
              isDisabled
              type="email"
              value={employeeEmail}
            />
          </FormGroup>
          <Controller
            name="alternateEmail"
            control={control}
            defaultValue={sessionStorage.getItem("alternateEmail") || ""}
            render={({ field, fieldState }) => (
              <FormGroup
                id="alernative-email"
                label="Alternate Email (optional)"
                helperText="Optional email for notifications"
                helperTextInvalid="This needs to be a valid email address"
                fieldId="alernative-email"
                validated={fieldState.error ? "error" : "default"}
              >
                <TextInput
                  aria-label="alernative-email"
                  onChange={field.onChange}
                  placeholder="example@mail.com"
                  type="email"
                  value={field.value}
                />
              </FormGroup>
            )}
            rules={{
              required: false,
              pattern: /[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}/g,
            }}
          />
          <FormGroup
            id="employee-agency"
            label="Agency"
            fieldId="profile-form-agency"
          >
            <TextInput
              aria-label="employee-agency"
              isDisabled
              type="text"
              value={employeeAgency}
            />
          </FormGroup>
          {employeeRole.localeCompare("approver") == 0 && (
            <Checkbox
              id="reviewer-checkbox"
              label="Red Hat VCS Reviewer"
              aria-label="disabled checked checkbox"
              defaultChecked
              isDisabled
            />
          )}
          <FormGroup fieldId="accept-test-buttons">
            <Button icon={<CheckCircleIcon />} variant="primary" type="submit">
              Accept
            </Button>
            <Button variant="plain" onClick={() => history.push("/")}>
              Back
            </Button>
          </FormGroup>
        </Form>
      </CardBody>
    </Card>
  );
};

export { Profile };
