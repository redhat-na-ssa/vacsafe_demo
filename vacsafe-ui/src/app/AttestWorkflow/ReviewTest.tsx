import * as React from 'react';
import {
  Button, 
  Card,
  CardFooter,
  CardBody,
  Form,
  FormGroup,
  TextInput
} from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import { OutlinedPaperPlaneIcon } from '@patternfly/react-icons';
import { useHistory } from 'react-router-dom';
import CardWithTitle from '@app/Components/CardWithTitle';
import { convertBase64ToBlob, dateConvert } from "@app/utils/utils";
import authaxios from "@app/utils/axiosInterceptor";
import { useState } from 'react';
import { ResponseErrorModal } from '@app/Components/FeedbackModals';
import { useKeycloak } from "@react-keycloak/web";

const ReviewTest: React.FunctionComponent = () => {
  const history = useHistory();
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false); 
  const [errorState, setErrorState] = useState('');
  const [isFormSubmitting, setIsFormSubmitting] = useState(false);
  const { keycloak } = useKeycloak();

  const testImage = sessionStorage.getItem("test_image_file") || "";
  const testImageFilename = sessionStorage.getItem("test_image_filename") || "";
  const employeeDOB = sessionStorage.getItem("dateOfBirth") || "";
  const altEmail = sessionStorage.getItem("alternateEmail") || "";
  const covidTestResult = sessionStorage.getItem("test_results") || "";
  const covidTestDate = sessionStorage.getItem("test_date") || "";
  
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeFirstName, setEmployeeFirstName] = useState("");
  const [employeeLastName, setEmployeeLastName] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [employeeAgency, setEmployeeAgency] = useState("");
  const [employeeDivision, setEmployeeDivision] = useState(""); 

  if(keycloak?.authenticated) {
    keycloak.loadUserProfile().then((profile) => {
      setEmployeeId(profile.attributes.workforceid[0]);
      setEmployeeName(profile.firstName + " " + profile.lastName);
      setEmployeeFirstName(profile.firstName || "");
      setEmployeeLastName(profile.lastName || "");
      setEmployeeEmail(profile.email || "");
      setEmployeeAgency(profile.attributes.agency[0]); 
      setEmployeeDivision(profile.attributes.division[0])
    })
  }

  const handleSubmit = ()=> {
    setIsFormSubmitting(true);
    const pamBaseUrl = window.REACT_APP_PAM_URL_BASE ? window.REACT_APP_PAM_URL_BASE : ""
    const attestUrl = pamBaseUrl + "/attestation/covid-test-result"
    const testCard = convertBase64ToBlob(testImage)
    const bodyFormData = new FormData();

    const employeeInfo = {
      "id": employeeId,
      "firstName": employeeFirstName,
      "lastName": employeeLastName,
      "dateOfBirth": employeeDOB,
      "email": employeeEmail,
      "alternateEmail": altEmail,
      "agencyCode": employeeAgency,
      "agencyName": employeeAgency,
      "divisionCode": employeeDivision
    }

    const testInfo = {
      "covidTestDate": covidTestDate,
      "covidTestResult": covidTestResult.toUpperCase()
    }

    bodyFormData.append('employee', new Blob([JSON.stringify(employeeInfo)], {type: "application/json"}) );
    bodyFormData.append('document', new Blob([JSON.stringify(testInfo)], {type: "application/json"}));
    bodyFormData.append('attachment', testCard);

    const headers = {
      'Content-Type': 'multipart/form-data;'
    }

    authaxios.post(attestUrl, bodyFormData, {
      headers: headers
    })
    .then(() => {
      setIsFormSubmitting(false);
      history.push('/thankyou');
    })
    .catch(res => {
      if(res.response) {
        setIsFormSubmitting(false);
        setErrorState(res.response.status);
        setIsErrorModalOpen(true);
      }
    })
  }

  return (
    <Card isRounded>
      <ResponseErrorModal 
        isModalOpen={isErrorModalOpen} 
        setIsModalOpen={setIsErrorModalOpen} 
        errorState={errorState}
      />
      <CardWithTitle
        title="Review Submission"
        info="Review your test result submission and validate that test result image matches displayed data. Please press Submit below to proceed."
      />
      <CardBody>
        <Form>
          <FormGroup label="Person ID" fieldId="review-test-form-employee-id">
            <TextInput
              aria-label="review-test-form-employee-id"
              isDisabled
              type="text"
              value={employeeId}
            />
          </FormGroup>
          <FormGroup
            fieldId="review-test-form-employee-full-name"
            label="Full Name"
          >
            <TextInput
              aria-label="review-test-form-employee-full-name"
              isDisabled
              type="text"
              value={employeeName}
            />
          </FormGroup>
          <FormGroup
            label="Date of Birth"
            fieldId="review-test-date-of-birth"
          >
            <TextInput
              aria-label="review-test-date-of-birth"
              isDisabled
              type="text"
              value={dateConvert(employeeDOB)}
            />
          </FormGroup>
          <FormGroup label="Test Results" fieldId="review-test-form-results">
            <TextInput
              aria-label="review-test-form-results"
              isDisabled
              type="text"
              value={sessionStorage.getItem("test_results") || ""}
            />
          </FormGroup>
          <FormGroup label="Test Date" fieldId="review-test-form-date">
            <TextInput
              aria-label="review-test-form-date"
              isDisabled
              type="text"
              value={dateConvert(sessionStorage.getItem("test_date") || "")}
            />
          </FormGroup>
          <FormGroup label="Alternative Email (optional)" fieldId="review-test-form-alternative-email">
            <TextInput
              aria-label="review-test-form-alternative-email"
              isDisabled
              type="text"
              value={altEmail}
            />
          </FormGroup>
          <FormGroup label="Image preview" fieldId="review-vax-image-preview">
            {testImage && testImageFilename.split('.').pop()?.toLowerCase() == "pdf" &&
              (typeof testImage !== "undefined" ? (
                <Button
                  icon={<ExternalLinkAltIcon />}
                  variant="link"
                  component="a"
                  href={URL.createObjectURL(convertBase64ToBlob(testImage))}
                  target="_blank"
                >
                  Test Result PDF
                </Button>
              ) : null)}
            {testImage && testImageFilename.split('.').pop()?.toLowerCase() != "pdf" &&
              (typeof testImage !== "undefined" ? (
                <div>
                  <img
                    src={URL.createObjectURL(convertBase64ToBlob(testImage))}
                    alt="test-results-image"
                  />
                </div>
              ) : null)}
          </FormGroup>
        </Form>
      </CardBody>
      <CardFooter>
        <Button
          icon={<OutlinedPaperPlaneIcon />}
          isDisabled={isFormSubmitting}
          isLoading={isFormSubmitting}
          onClick={handleSubmit}
          variant="primary"
        >
          Submit
        </Button>
        <Button onClick={() => history.push("/attest/test")} variant="plain">
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { ReviewTest };
