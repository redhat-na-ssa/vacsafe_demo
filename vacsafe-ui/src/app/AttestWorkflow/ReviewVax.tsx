import React, { useState } from "react";
import {
  Card,
  CardFooter,
  CardBody,
  Button,
  Form,
  FormGroup,
  TextInput
} from "@patternfly/react-core";
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import { OutlinedPaperPlaneIcon } from "@patternfly/react-icons";
import { useHistory } from "react-router-dom";
import { convertBase64ToBlob, dateConvert } from "@app/utils/utils";
import authaxios from "@app/utils/axiosInterceptor";
import { ResponseErrorModal } from '@app/Components/FeedbackModals';
import { useKeycloak } from "@react-keycloak/web";
import CardWithTitle from "@app/Components/CardWithTitle";

const ReviewVax: React.FunctionComponent = () => {
  const [isErrorModalOpen, setIsErrorModalOpen] = useState(false);
  const [errorState, setErrorState] = useState("");
  const [isFormSubmitting, setIsFormSubmitting] = useState(false);
  const employeeDOB = sessionStorage.getItem("dateOfBirth") || "";
  const vaccine = sessionStorage.getItem("vaccine") || "";
  const shotDate1 = sessionStorage.getItem("vax_date_1") || "";
  const shotDate2 = sessionStorage.getItem("vax_date_2") || "";
  const vaxImage = sessionStorage.getItem("vax_image") || "";
  const vaxFilename = sessionStorage.getItem("vax_image_filename") || "";
  const altEmail = sessionStorage.getItem("alternateEmail") || "";
  const history = useHistory();
  const { keycloak } = useKeycloak();

  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");
  const [employeeFirstName, setEmployeeFirstName] = useState("");
  const [employeeLastName, setEmployeeLastName] = useState("");
  const [employeeEmail, setEmployeeEmail] = useState("");
  const [employeeAgency, setEmployeeAgency] = useState("");
  const [employeeDivision, setEmployeeDivision] = useState(""); 

  if(keycloak?.authenticated) {
    keycloak.loadUserProfile().then((profile) => {
      setEmployeeId(profile.attributes.workforceid[0]);
      setEmployeeName(profile.firstName + " " + profile.lastName);
      setEmployeeFirstName(profile.firstName || "");
      setEmployeeLastName(profile.lastName || "");
      setEmployeeEmail(profile.email || "");
      setEmployeeAgency(profile.attributes.agency[0]); 
      setEmployeeDivision(profile.attributes.division[0])
    })
  }

  const formSubmitSuccess = () => {
    setIsFormSubmitting(false);
    history.push("/thankyou");
  }
  const handleSubmit = () => {
    setIsFormSubmitting(true);
    const pamBaseUrl = window.REACT_APP_PAM_URL_BASE ? window.REACT_APP_PAM_URL_BASE : "";
    const attestUrl = pamBaseUrl + "/attestation/vax";
    const vaxCard = convertBase64ToBlob(vaxImage);
    const bodyFormData1 = new FormData();
    const bodyFormData2 = new FormData();

    const employeeInfo = {
      id: employeeId,
      firstName: employeeFirstName,
      lastName: employeeLastName,
      dateOfBirth: employeeDOB,
      email: employeeEmail,
      alternateEmail: altEmail,
      agencyCode: employeeAgency,
      agencyName: employeeAgency,
      divisionCode: employeeDivision,
    };

    const vaxInfo1 = {
      vaccineBrand: vaccine.toUpperCase(),
      vaccineAdministrationDate: shotDate1,
      vaccineShotNumber: 1,
    };

    const vaxInfo2 = {
      vaccineBrand: vaccine.toUpperCase(),
      vaccineAdministrationDate: shotDate2,
      vaccineShotNumber: 2,
    };

    bodyFormData1.append(
      "employee",
      new Blob([JSON.stringify(employeeInfo)], { type: "application/json" })
    );
    bodyFormData1.append(
      "document",
      new Blob([JSON.stringify(vaxInfo1)], { type: "application/json" })
    );
    bodyFormData1.append("attachment", vaxCard);

    bodyFormData2.append(
      "employee",
      new Blob([JSON.stringify(employeeInfo)], { type: "application/json" })
    );
    bodyFormData2.append(
      "document",
      new Blob([JSON.stringify(vaxInfo2)], { type: "application/json" })
    );
    bodyFormData2.append("attachment", vaxCard);

    const headers = {
      "Content-Type": "multipart/form-data;",
    };

    authaxios.post(attestUrl, bodyFormData1, {
        headers: headers
      })
      .then(() => {
        if (shotDate2) {
          authaxios.post(attestUrl, bodyFormData2, {
              headers: headers
            })
            .then(() => {
              formSubmitSuccess();
            })
            .catch((res) => {
              if (res.response) {
                setIsFormSubmitting(false);
                setErrorState(res.response.status);
                setIsErrorModalOpen(true);
              }
            });
        } else {
          formSubmitSuccess();
        }
      })
      .catch((res) => {
        if (res.response) {
          setIsFormSubmitting(false);
          setErrorState(res.response.status);
          setIsErrorModalOpen(true);
        }
      });
  }

  return (
    <Card isRounded>
      <ResponseErrorModal 
        isModalOpen={isErrorModalOpen} 
        setIsModalOpen={setIsErrorModalOpen} 
        errorState={errorState}
      />
      <CardWithTitle
        title="Review Vaccination Submission"
        info="Please review your vaccination record submission and validate that card image matches displayed data or it will be rejected. Please press Submit below to proceed."
      />
      <CardBody>
        <Form>
          <FormGroup label="Person ID" fieldId="review-vax-form-employee-id">
            <TextInput
              aria-label="review-vax-form-employee-id"
              isDisabled
              type="text"
              value={employeeId}
            />
          </FormGroup>
          <FormGroup
            fieldId="review-vax-form-employee-full-name"
            label="Full Name"
          >
            <TextInput
              aria-label="review-vax-form-employee-full-name"
              isDisabled
              type="text"
              value={employeeName}
            />
          </FormGroup>
          <FormGroup
            label="Date of Birth"
            fieldId="review-vax-form-date-of-birth"
          >
            <TextInput
              aria-label="review-vax-form-date-of-birth"
              isDisabled
              type="text"
              value={dateConvert(employeeDOB)}
            />
          </FormGroup>
          <FormGroup label="Vaccine" fieldId="review-vax-form-vaccine-brand">
            <TextInput
              aria-label="review-vax-form-vaccine-brand"
              isDisabled
              type="text"
              value={
                vaccine.localeCompare("Johnson") ? vaccine : "Johnson & Johnson"
              }
            />
          </FormGroup>
          <FormGroup
            label="Date of Shot 1"
            fieldId="review-vax-form-shot-1-date"
          >
            <TextInput
              aria-label="review-vax-form-shot-1-date"
              isDisabled
              type="text"
              value={dateConvert(shotDate1)}
            />
          </FormGroup>
          <FormGroup
            label="Date of Shot 2 (if applicable)"
            fieldId="review-vax-form-shot-2-date"
          >
            <TextInput
              aria-label="review-vax-form-shot-2-date"
              isDisabled
              type="text"
              value={dateConvert(shotDate2)}
            />
          </FormGroup>
          <FormGroup
            label="Alternative Email (optional)"
            fieldId="review-alternative-email"
          >
            <TextInput
              aria-label="review-vax-form-alternative-email"
              isDisabled
              type="text"
              value={altEmail}
            />
          </FormGroup>
          <FormGroup label="Image preview" fieldId="review-vax-image-preview">
            {vaxImage && vaxFilename.split('.').pop()?.toLowerCase() == "pdf" && (typeof vaxImage !== "undefined" ? (
                <Button
                  icon={<ExternalLinkAltIcon />}
                  variant="link"
                  component="a"
                  href={URL.createObjectURL(convertBase64ToBlob(vaxImage))}
                  target="_blank"
                >
                  Vaccination Card PDF
                </Button>
            ) : null)}
            {vaxImage && vaxFilename.split('.').pop()?.toLowerCase() != "pdf" &&
              (typeof vaxImage !== "undefined" ? (
                <div>
                  <img
                    src={URL.createObjectURL(convertBase64ToBlob(vaxImage))}
                    alt="vaccination-card"
                  />
                </div>
              ) : null)}
          </FormGroup>
        </Form>
      </CardBody>
      <CardFooter>
        <Button
          icon={<OutlinedPaperPlaneIcon />}
          isDisabled={isFormSubmitting}
          isLoading={isFormSubmitting}
          variant="primary"
          onClick={handleSubmit}
        >
          Submit
        </Button>
        <Button variant="plain" onClick={() => history.push("/attest/vax")}>
          Back
        </Button>
      </CardFooter>
    </Card>
  );
};

export { ReviewVax };
