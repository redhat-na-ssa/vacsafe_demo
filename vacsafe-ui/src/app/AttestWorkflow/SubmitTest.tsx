import React, { useState } from "react";
import {
  Card,
  CardBody
} from "@patternfly/react-core";
import { useHistory } from "react-router-dom";
import { convertBase64ToBlob, convertBlobToBase64, dateConvert } from "@app/utils/utils";
import CardWithTitle from "@app/Components/CardWithTitle";
import { useKeycloak } from "@react-keycloak/web";
import CovidTestForm from "@app/Components/CovidTestForm";

const SubmitTest: React.FunctionComponent = () => {
  const history = useHistory();
  const { keycloak } = useKeycloak();
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");

  if(keycloak?.authenticated) {
    keycloak.loadUserProfile().then((profile) => {
      setEmployeeId(profile.attributes.workforceid[0]);
      setEmployeeName(profile.firstName + " " + profile.lastName);
    })
  }

  const onLoad = () => {
    const data = sessionStorage.getItem("test_image_file") || "";

    if (data !== "") {
      return new File([convertBase64ToBlob(data)], sessionStorage.getItem("test_image_filename") || "");
    }

    return "";
  }

  const onSubmit = data => {
    sessionStorage.setItem("test_results", data["test_results"]);
    sessionStorage.setItem("test_date", dateConvert(data["test_date"]));
    sessionStorage.setItem("test_image_filename", data["test_image_file"].name)
    convertBlobToBase64(data["test_image_file"]).then(base64 => {
      sessionStorage["test_image_file"] = base64;
    }).then(() => history.push("/attest/review/test"));
  };

  return (
    <Card isRounded>
      <CardWithTitle title="Test Results" info="Submit Test Results" />
      <CardBody>
        <CovidTestForm
          isHRSubmit={false}
          employeeId={employeeId}
          employeeName={employeeName}
          onSubmit={onSubmit}
          onLoad={onLoad}
        />
      </CardBody>
    </Card>
  );
};

export { SubmitTest };