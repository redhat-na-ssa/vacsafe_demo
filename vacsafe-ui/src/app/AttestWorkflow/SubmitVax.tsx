import React, { useState } from "react";
import {
  Card,
  CardBody
} from "@patternfly/react-core";
import { useHistory } from "react-router-dom";
import { convertBase64ToBlob, convertBlobToBase64, dateConvert } from "@app/utils/utils";
import { useKeycloak } from "@react-keycloak/web";
import CardWithTitle from "@app/Components/CardWithTitle";
import CovidVaxForm from "@app/Components/CovidVaxForm";

const SubmitVax: React.FunctionComponent = () => {
  const history = useHistory();
  const { keycloak } = useKeycloak();
  const [employeeId, setEmployeeId] = useState("");
  const [employeeName, setEmployeeName] = useState("");

  if(keycloak?.authenticated) {
    keycloak.loadUserProfile().then((profile) => {
      setEmployeeId(profile.attributes.workforceid[0]);
      setEmployeeName(profile.firstName + " " + profile.lastName);
    })
  }

  const onLoad = () => {
    const data = sessionStorage.getItem("vax_image") || "";

    if (data !== "") {
      return new File([convertBase64ToBlob(data)], sessionStorage.getItem("vax_image_filename") || "");
    }

    return "";
  }

  const onSubmit = (data) => {
    sessionStorage.setItem("vaccine", data["vaccine"]);
    sessionStorage.setItem("vax_date_1", dateConvert(data["vax_date_1"]));
    sessionStorage.setItem("vax_date_2", dateConvert(data["vax_date_2"]));
    sessionStorage.setItem("vax_image_filename", data["vax_image"].name)
    convertBlobToBase64(data["vax_image"]).then((base64) => {
      sessionStorage["vax_image"] = base64;
    }).then(()=> history.push("/attest/review/vax"));
  };

  return (
    <Card isRounded>
      <CardWithTitle title="Vaccination Form" info="Enter vaccination info" />
      <CardBody>
        <CovidVaxForm 
          isHRSubmit={false}
          employeeId={employeeId}
          employeeName={employeeName}
          onSubmit={onSubmit}
          onLoad={onLoad}
        />
      </CardBody>
    </Card>
  );
};

export { SubmitVax };
