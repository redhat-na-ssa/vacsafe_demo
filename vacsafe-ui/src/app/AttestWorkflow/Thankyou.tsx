import * as React from 'react';
import { Button, Card, CardFooter } from "@patternfly/react-core";
import { WindowCloseIcon } from '@patternfly/react-icons';
import CardWithTitle from '@app/Components/CardWithTitle';
import {useKeycloak} from "@react-keycloak/web";

const Thankyou: React.FunctionComponent = () => {
  const {keycloak} = useKeycloak();
  return (
    <Card isRounded>
      <CardWithTitle
        title="Thank You!"
        info="Thank you for your submission. Your documents have been received by Human Resources."
      />
      <CardFooter>
        <Button
          icon={<WindowCloseIcon />}
          onClick={() => {
            keycloak.logout({redirectUri: location.origin});
            sessionStorage.clear()
          }}
          type="button"
          variant="danger"
        >
          Finish and Log Out
        </Button>
      </CardFooter>
    </Card>
  );
}

export { Thankyou };
