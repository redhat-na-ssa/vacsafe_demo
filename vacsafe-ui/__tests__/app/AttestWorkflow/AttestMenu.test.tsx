import React from 'react';
import { shallow } from 'enzyme';
import { AttestMenu } from '../../../src/app/AttestWorkflow/AttestMenu';

const mocked_keycloak = require('../../../__mocks__/keycloakMock');

jest.mock('@react-keycloak/web', () => ({
    useKeycloak: jest.fn( ()=>({keycloak: mocked_keycloak}))
}))

describe('AttestWorkflow - AttestMenu', ()=> {
    it('should be defined', () => {
        expect(AttestMenu).toBeDefined()
    })
    it("should render without crashing", () => {
        const wrapper = shallow(<AttestMenu />)
        expect(wrapper)
    })
    it("should match existing snapshot", () => {
        const wrapper = shallow(<AttestMenu />)
        expect(wrapper).toMatchSnapshot()
    })
})
