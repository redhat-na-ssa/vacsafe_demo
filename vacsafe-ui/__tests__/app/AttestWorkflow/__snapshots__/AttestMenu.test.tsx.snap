// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AttestWorkflow - AttestMenu should match existing snapshot 1`] = `
<Card
  isRounded={true}
>
  <CardWithTitle
    info="Please select from the following:"
    title="Menu"
  />
  <CardBody>
    <Stack
      hasGutter={true}
    >
      <Button
        icon={
          <FileUploadIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        }
        id="submit-vax-card-menu-button"
        isSmall={true}
        onClick={[Function]}
        variant="tertiary"
      >
        <Text>
          Submit Vaccination Record
        </Text>
        <Text>
          Supporting Document Upload Required
        </Text>
      </Button>
      <Button
        icon={
          <FileUploadIcon
            color="currentColor"
            noVerticalAlign={false}
            size="sm"
          />
        }
        id="submit-vax-test-menu-button"
        isSmall={true}
        onClick={[Function]}
        variant="tertiary"
      >
        <Text>
          Submit COVID Test Results
        </Text>
        <Text>
          Supporting Document Upload Required
        </Text>
      </Button>
    </Stack>
  </CardBody>
  <CardFooter>
    <Button
      onClick={[Function]}
      variant="plain"
    >
      Back
    </Button>
  </CardFooter>
</Card>
`;
